src/pool/srv_pool.c

```diff
@@ -7316,10 +7313,10 @@ ds_pool_extend_handler(crt_rpc_t *rpc)
 	if (rc != 0)
 		goto out;
 
-	rc = pool_svc_update_map(svc, pool_opc_2map_opc(opc_get(rpc->cr_opc)),
-				 false /* exclude_rank */, &rank_list, domains, ndomains,
-				 NULL, NULL, &out->peo_op.po_map_version,
-				 &out->peo_op.po_hint, MUS_DMG);
+	rc =
+	    pool_svc_update_map(svc, pool_opc_2map_opc(opc_get(rpc->cr_opc)),
+				false /* exclude_rank */, &rank_list, domains, ndomains, NULL, NULL,
+				&out->peo_op.po_map_version, &out->peo_op.po_hint, MUS_DMG, false);
```

Comment:
POOL_EXTEND seems will never cause RF broken so last parameter can be true.
If you check opc == EXCLUDE inside pool_svc_update_map_internal then either is fine.

Reply:
A second think...It might be ok to proceed, in the broken case, rebuild might hang until restarting dead engines..Let me change it to true always.